File: SPCode/CS/UI/Dialog/SPFileDialog.cs
using System.Text.RegularExpressions;
using System.Diagnostics;
using System;
using System.Drawing.Drawing2D;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Collections.Generic;
using WeifenLuo.WinFormsUI;
using Microsoft.Win32;
using WeifenLuo;


namespace SoftLogik.Win
{
	namespace UI
	{
		public partial class SPFileDialog
		{
			
			
			public enum FileDialogTypes
			{
				General = 0,
				Picture,
				OfficeDocument,
				Video,
				Audio
				
			}
			
			public static string ShowDialog(FileDialogTypes FileType)
			{
				
				try
				{
					using (OpenFileDialog myFileDialog = new OpenFileDialog())
					{
						myFileDialog.Filter = GetFilters(FileType);
						if (myFileDialog.ShowDialog() == DialogResult.OK)
						{
							return myFileDialog.FileName;
						}
						
					}
					
					
				}
				catch (Exception)
				{
				}
				return null;
			}
			public static string[] ShowMultiDialog(FileDialogTypes FileType)
			{
				
				try
				{
					using (OpenFileDialog myFileDialog = new OpenFileDialog())
					{
						if (myFileDialog.ShowDialog() == DialogResult.OK)
						{
							return myFileDialog.FileNames;
						}
						
					}
					
					
				}
				catch (Exception)
				{
				}
				return null;
			}
			
			private static string GetFilters(FileDialogTypes filetype)
			{
				switch (filetype)
				{
					case FileDialogTypes.Picture:
						return "All Picture Files(*.jpeg,*.jpe,*.jpg,*.gif,*.png,*.bmp,*.wmf,*.ico)|*.jpeg;*.jpe;*.jpg;*.gif;*.png;*.bmp;*.wmf;*.ico|JPEG Files(*.jpg,*.jpe,*.jpeg)|*.jpg;*.jpe;*.jpeg";
					default:
						return "All Files(*.*)|*.*";
				}
				
				return null;
			}
		}
	}
	
	
}

File: SPCode/UI/Dialog/SPFileDialog.vb

Namespace UI
    Public Class SPFileDialog

        Public Enum FileDialogTypes As Integer
            General = 0
            Picture
            OfficeDocument
            Video
            Audio

        End Enum

        Public Shared Function ShowDialog(ByVal FileType As FileDialogTypes) As String

            Try
                Using myFileDialog As New OpenFileDialog
                    With myFileDialog
                        .Filter = GetFilters(FileType)
                        If .ShowDialog() = DialogResult.OK Then
                            Return .FileName
                        End If
                    End With

                End Using

            Catch ex As Exception
            End Try
            Return Nothing
        End Function
        Public Shared Function ShowMultiDialog(ByVal FileType As FileDialogTypes) As String()

            Try
                Using myFileDialog As New OpenFileDialog
                    With myFileDialog
                        If .ShowDialog() = DialogResult.OK Then
                            Return .FileNames
                        End If
                    End With

                End Using

            Catch ex As Exception
            End Try
            Return Nothing
        End Function

        Private Shared Function GetFilters(ByVal filetype As FileDialogTypes) As String
            Select Case filetype
                Case FileDialogTypes.Picture
                    Return "All Picture Files(*.jpeg,*.jpe,*.jpg,*.gif,*.png,*.bmp,*.wmf,*.ico)|*.jpeg;*.jpe;*.jpg;*.gif;*.png;*.bmp;*.wmf;*.ico|JPEG Files(*.jpg,*.jpe,*.jpeg)|*.jpg;*.jpe;*.jpeg"
                Case Else
                    Return "All Files(*.*)|*.*"
            End Select

            Return Nothing
        End Function
    End Class
End Namespace

